os/pm/pm_procfs.c

```diff
@@ -466,24 +479,58 @@ static size_t power_lock_write(FAR struct file *filep, FAR const char *buffer, s
 /****************************************************************************
  * Name: power_unlock_write
  ****************************************************************************/
+#ifdef CONFIG_ARCH_CORTEXA32
+static size_t power_unlock_write(FAR struct file *filep, FAR const char *buffer, size_t timer_interval)
+#else
```

Comment:
I think when we eliminated the need to PM control from APP size, we also eliminated the need for having option to set wakeup timer.
We do not need to provide customizable timer option now. Core will wakeup from interrupt, or from fixed timer set for wifi alive signal.

Reply:
As per the conversation we had before, SW timer can be an option to choose whether to apply or not, if yes, it will be an extra interrupt signal, if there is no other signal arrive during the time period given. Thus, it should have no conflict even though we don't allow direct state jump to PM_SLEEP.
Could you please elaborate more on "fixed timer set for wifi alive signal"?

Reply:
Hi @amogh-samsung ,

Okay, thanks for reminding this part. But I will still leave the timer interrupt setup, just in case there is some scenario that might require the timer in the future. If the application dont need it, you can just simply provide 0 to the last argument when relaxing the state, so it will not trigger any timer interrupt.

Reply:
I get your point, it can be a good to have feature.
I have one question though, if multiple applications block and then unblock the same domain, then the timer interval will be set by the last unblocking application right?

Reply:
Yes, you are right~ If there is some other requirement to choose the timer interval from multiple application in the future, we can extend that too.

Reply:
Ohkay, we'll also take @sunghan-chang 's opinion on this please